### PR TITLE
Backport decodeme crate to v9.1.x

### PR DESCRIPTION
**Pull request by michaelwoerister**
This PR adds the `decodeme` crate to 9.1.x so that we can use it to make 10.x+ releases backwards compatible with the data format used in 9.x versions.

See https://github.com/rust-lang/measureme/issues/178 for more information.

r? @rylev & @wesleywiser

### CODE REVIEW

**Comment by michaelwoerister** [comment]
Hm, I think technically the changes in here are not backwards compatible with `9.1.2` because `LightweightEvent` has changed. So maybe the crate version should not be `9.1.2` but something else? Maybe something unstable such as `0.9.0-decodeme-backport`?

File: analyzeme/src/profiling_data.rs [hunk 1]
```diff
@@ -261,6 +169,11 @@ impl ProfilingDataBuilder {
         )
         .unwrap();
 
+        string_table.alloc_metadata(&*format!(
```
Comment (rylev) [inline analyzeme/src/profiling_data.rs:172]: What's this for?

File: analyzeme/src/profiling_data.rs [hunk 2]
```diff
@@ -82,97 +44,43 @@ impl ProfilingData {
...
+        self.event_decoder.decode_full_event(event_index)
+    }
 
-        LightweightEvent {
-            data: self,
-            event_index,
-            timestamp,
-            thread_id: raw_event.thread_id,
-        }
+    fn decode_lightweight_event<'a>(&'a self, event_index: usize) -> LightweightEvent {
```
Comment (rylev) [inline analyzeme/src/profiling_data.rs:82]: nit: remove the lifetime

**Review by rylev — APPROVED** [review]
Looks good! I'm not sure how I feel about the backport strategy. Effectively we're making a new release of of v9.1 so that 10.0 can use that as a snapshot even though we don't intend for users to use the new version of 9.1. This feels strange.

Technically to be semver compliant we'd need to release this snapshot as 10.0 and then release 11.0 (since both introduce breaking changes).

File: analyzeme/src/profiling_data.rs (hunk 1 above)
Comment (michaelwoerister) [reply analyzeme/src/profiling_data.rs:172]: The string table contains a "metadata" entry that stores some global information. The `start_time` in there is used for mapping the timings in each event to a `SystemTime`, so we need it for test cases that look at those times.

**Comment by michaelwoerister** [comment]
Closing in favor of https://github.com/rust-lang/measureme/pull/180.

**Pull request closed without merging**